### PR TITLE
Bumps retention of predictions to 365 days

### LINKED ISSUE #3582: MoreCast: Data Retention Policy; 1 Year
**As a** Forecaster

**I need** to keep weather data for the entire past fire season

**So That** I can analyze historical model data

**Acceptance Criteria**
- [ ] Keep model prediction data for 365 days instead of the current 21
- [ ] Don't add more data to the materialized view as it could impact performance

**Definition of Done**
https://github.com/bcgov/wps/wiki/Definition-of-Done

### PR DESCRIPTION
**Pull request by conbrad**
closes #3582 
# Test Links:
[Landing Page](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3597-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Comment by conbrad** [comment]
Closing while we stabilize

File: api/alembic/versions/025a81a4b7bd_limit_mat_view_to_21_days.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+depends_on = None
+
+
+def upgrade():
+    op.create_index(
+    op.f('ix_wsm_prediction_timestamp_station_code'),
+    'weather_station_model_predictions',
+    ['prediction_timestamp', 'station_code'],
+    unique=False
+)
```
Comment (brettedw) [inline api/alembic/versions/025a81a4b7bd_limit_mat_view_to_21_days.py:24]: Seems to significantly speed up the mat view query/refresh. Might be useful with `weather_station_model_predictions` growing

File: api/alembic/versions/025a81a4b7bd_limit_mat_view_to_21_days.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            weather_station_model_predictions.prediction_timestamp) = 20
+        GROUP BY
+            weather_station_model_predictions.station_code,
+            date(weather_station_model_predictions.prediction_timestamp)
+                ) AS latest
+                ON
+        weather_station_model_predictions.prediction_timestamp = latest.latest_prediction
+        AND weather_station_model_predictions.station_code = latest.station_code
+    WHERE
+        weather_station_model_predictions.prediction_timestamp >= current_date  - INTERVAL '21 days'
```
Comment (conbrad) [inline api/alembic/versions/025a81a4b7bd_limit_mat_view_to_21_days.py:72]: Not sure if this matters, but extra space here:

```suggestion
        weather_station_model_predictions.prediction_timestamp >= current_date - INTERVAL '21 days'
```

**Review by dgboss — APPROVED** [review]

**Comment by dgboss** [comment]
Looks great!

**Pull request merged**